crates/revm/src/optimism/executor.rs

```diff
@@ -0,0 +1,313 @@
+                if transaction.is_deposit() && !result.is_success() {
+                    cumulative_gas_used += transaction.gas_limit();
+                } else if is_regolith || !transaction.is_deposit() {
+                    cumulative_gas_used += result.gas_used()
+                } else if transaction.is_deposit() && !transaction.is_system_transaction() {
+                    cumulative_gas_used += transaction.gas_limit();
+                }
+
+                // Pay out fees to Optimism vaults if the transaction is not a deposit. Deposits
+                // are exempt from vault fees.
+                if !transaction.is_deposit() {
+                    // Route the l1 cost and base fee to the appropriate optimism vaults
+                    if let Some(l1_cost) = l1_cost {
+                        self.increment_account_balance(
+                            block.number,
+                            super::l1_cost_recipient(),
+                            l1_cost,
+                            &mut post_state,
+                        )?
+                    }
+                    self.increment_account_balance(
+                        block.number,
+                        super::base_fee_recipient(),
+                        U256::from(
+                            block
+                                .base_fee_per_gas
+                                .unwrap_or_default()
+                                .saturating_mul(result.gas_used()),
+                        ),
+                        &mut post_state,
+                    )?;
+                }
+            } else {
+                cumulative_gas_used += result.gas_used();
+            }
+
+            // cast revm logs to reth logs
+            let logs = result.logs().into_iter().map(into_reth_log).collect();
+
+            // Push transaction changeset and calculate header bloom filter for receipt.
+            post_state.add_receipt(
+                block.number,
+                Receipt {
+                    tx_type: transaction.tx_type(),
+                    // Success flag was added in `EIP-658: Embedding transaction status code in
+                    // receipts`.
+                    success: result.is_success(),
+                    cumulative_gas_used,
+                    logs,
+                    // Deposit nonce is only recorded after Regolith for deposit transactions.
+                    deposit_nonce: (is_regolith && transaction.is_deposit())
+                        .then_some(sender_account.info.nonce),
+                },
+            );
+
+            // Reset all revm configuration flags for the next iteration.
+            if transaction.is_deposit() {
+                self.evm.env.cfg.disable_base_fee = false;
+                self.evm.env.cfg.disable_block_gas_limit = false;
+                self.evm.env.cfg.disable_balance_check = false;
+                self.evm.env.cfg.disable_gas_refund = false;
+            }
```

Comment:
It seems like we should reset it to whatever was in the env before overwriting earlier in the function - mainly for testability purposes.